doc/guides/dts/index.rst

```diff
@@ -346,42 +346,113 @@ Below is sample header content generated for this node, in
-binding for the node (``nxp,kinetis-sim``) and the node's unit address
-(``...@4004700``).
+All macros generated from devicetree start with ``DT_`` and use all-uppercase.
 
-The ``*_BASE_ADDRESS`` part of the identifier is a fixed identifier generated
-from the special ``reg`` property on the node. ``*_SIZE`` is also generated
-from ``reg``. Other suffixes, like ``*_LABEL``, are generated directly from the
-property name.
+Node prefixes
+-------------
 
-The second macro (``DT_INST_0_NXP_KINETIS_SIM_BASE_ADDRESS``) is an alias for
-the first macro. The node identifier ``...INST_0_NXP_KINETIS_SIM_...`` means
-"the first node with compatible string ``nxp,kinetis-sim``.
+Macros generated from particular devicetree nodes start with ``DT_<node>``,
+where ``<node>`` identifies the node. Several prefixes get generated for nodes:
 
-Aliases are also generated from any properties in the ``/aliases`` node. Take
-the DTS fragment below as an example.
+``DT_<compatible>_<unit-address>``
+    The compatible string for the node followed by its unit address.
+
+    For nodes that don't have a unit address (no ``...@<address>`` in the node
+    name), the unit address of the parent node is used instead. If the parent
+    node has no unit address either, the name of the node is used for
+    ``<unit-address>``, as a fallback.
+
+    For the ``sim@40047000`` node above, the compatible string is
+    ``"nxp,kinetis-sim"``. This gets converted to ``NXP_KINETIS_SIM`` by
+    uppercasing and replacing non-alphanumeric characters with underscores.
+    Adding the unit address gives the node prefix
+    ``DT_NXP_KINETIS_SIM_40047000``.
+
+``DT_INST_<instance-no.>_<compatible>``
+    The compatible string for the node together with an instance number.
+
+    The instance number is a unique index among all enabled
+    (``status = "okay"``) nodes that have a particular compatible string,
+    counting from zero. For example, if there are two enabled nodes that have
+    ``compatible = "foo,uart"``, then these node prefixes get generated:
+
+    .. code:: none
+
+        DT_INST_0_FOO_UART
+        DT_INST_1_FOO_UART
+
+``DT_ALIAS_<alias>``
+    Generated from the names of any properties in the ``/aliases`` node that
+    point to the node.
+
+    For example, assume ``/aliases`` looks like this:
+
+    .. code-block:: none
+
+       aliases {
+               uart-1 = &uart1;
+       };
+
+    The alias name (``i2c-1``) is converted to ``I2C_1`` by uppercasing and
+    replacing non-alphanumeric characters with underscores, generating the node
+    prefix ``DT_ALIAS_UART_1`` for the ``&uart1`` node.
+
+    .. note::
+
+       Currently, an older deprecated ``DT_<compatible>_<alias>`` form is also
+       generated for aliases. For the example above, assuming the compatible
+       string for the ``&uart1`` node is ``"foo,uart"``, this gives
+       ``DT_FOO_UART_UART_1``.
+
+       Work is underway to replace this form with ``DT_ALIAS_*``.
+
+Property suffixes
```

Comment:
Node Property suffixes?

Reply:
Might be clear from context. `Node prefixes`/`Node property suffixes` seems a bit weird.